src/libostree/ostree-repo-checkout.c

```diff
@@ -1388,10 +1388,18 @@ ostree_repo_checkout_at (OstreeRepo                        *self,
 
   g_autoptr(GFile) target_dir = NULL;
 
-  if (strcmp (options->subpath, "/") != 0)
-    target_dir = g_file_get_child (commit_root, options->subpath);
-  else
+  if (strcmp (options->subpath, "/") != 0) {
+    if (options->subpath[0] == '/') {
+      g_autoptr(GFile) abs_root_dir = g_file_new_for_path ("/");
+      g_autoptr(GFile) subpath = g_file_new_for_path (options->subpath);
+      g_autofree char* rel_subpath = g_file_get_relative_path (abs_root_dir, subpath);
+      target_dir = g_file_get_child (commit_root, rel_subpath);
+    } else {
+      target_dir = g_file_get_child (commit_root, options->subpath);
```

Comment:
I don't think this is right. The second parameter to g_file_get_child() should be a basename, i.e. it should surely not contain the path separator /.

Reply:
So I suppose we need to use `g_file_resolve_relative_path`.

Reply:
@mcatanzaro I have changed the code. Could you tell me if I am now using correctly glib api?

Reply:
Well... the new version doesn't make sense either. g_path_skip_root() assumes you are starting from an absolute path, then g_file_resolve_relative_path() converts it back to an absolute path. So that doesn't make sense, right?

Reply:
It is not the same absolute path. We compose for example `/root/path` with `/sub/path` and we expect to get `/root/path/sub/path`.
However, I have noticed that `g_path_skip_root` returns NULL on relative path and I believe we need to accept `sub/path` as subpath to behave like `/sub/path`.

Reply:
> It is not the same absolute path. We compose for example /root/path with /sub/path and we expect to get /root/path/sub/path.

Ack!

Reply:
I have looked more. The commit root path is a ostree specific implementation of GFile. And it is considered as the root. So passing an absolute or relative path as parameter will resolve to the same. So we should just switch `g_file_get_child` to `g_file_resolve_relative_path` without any other changes.